drodrpg/DRODLib/TotalMapStates.cpp

```diff
@@ -0,0 +1,142 @@
+/* ***** BEGIN LICENSE BLOCK *****
+ * Version: MPL 1.1
+ *
+ * The contents of this file are subject to the Mozilla Public License Version
+ * 1.1 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * http://www.mozilla.org/MPL/
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+ * for the specific language governing rights and limitations under the
+ * License.
+ *
+ * The Original Code is Deadly Rooms of Death.
+ *
+ * The Initial Developer of the Original Code is
+ * Caravel Software.
+ * Portions created by the Initial Developer are Copyright (C) 2025
+ * Caravel Software. All Rights Reserved.
+ *
+ * Contributor(s):
+ * Kieran Millar
+ *
+ * ***** END LICENSE BLOCK ***** */
+
+ //TotalMapStates.cpp.
+ //Implementation of CTotalMapStates.
+
+#include "TotalMapStates.h"
+#include "DbSavedGames.h"
+#include <BackEndLib/Types.h>
+
+#include <map>
+
+//*****************************************************************************
+//
+//Public methods.
+//
+
+//*****************************************************************************
+//Constructor
+CTotalMapStates::CTotalMapStates()
+{
+	this->mapStates = map<UINT, MapState>();
```

Comment:
You don't need this line; mapStates is automatically default constructed to an empty map.